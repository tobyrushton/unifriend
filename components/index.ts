export * from './providers'
export * from './screens'
