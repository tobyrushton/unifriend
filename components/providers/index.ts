export * from './LoadingProvider'
export * from './UserProvider'
export * from './NotificationProvider'
