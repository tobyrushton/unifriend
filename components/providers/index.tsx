export * from './LoadingProvider'
export * from './UserProvider'