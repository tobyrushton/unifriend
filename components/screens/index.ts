export * from './LoadingScreen'
export * from './AuthScreen'
export * from './ConfirmEmailScreen'
export * from './ForgottenPasswordScreen'
