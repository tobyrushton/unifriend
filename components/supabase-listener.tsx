'use client'

import { useRouter } from 'next/navigation'
import { useEffect, FC } from 'react'
import { supabase } from '../lib/supabase'

export const SupabaseListener: FC<{ accessToken?: string }> = ({
    accessToken,
}) => {
    const router = useRouter()

    useEffect(() => {
        supabase.auth.onAuthStateChange((_event, session) => {
            if (session?.access_token !== accessToken) {
                router.refresh()
            }
        })
    }, [accessToken, router])

    return null
}
