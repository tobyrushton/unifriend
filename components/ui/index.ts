'use client'

export * from './Logo'
export * from './Text'
export * from './Button'
export * from './Input'
export * from './Exit'
export * from './Notification'
export * from './Navbar'
export * from './DropDown'
export * from './ProfilePicture'
export * from './Toggle'
export * from './Message'
export * from './User'
