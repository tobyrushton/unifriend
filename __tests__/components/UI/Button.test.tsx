import { render, fireEvent } from '@testing-library/react'
import { Button } from '../../../components'
import '@testing-library/jest-dom'

describe('Button', () => {
    it('should render', () => {
        const { getByText, container } = render(
            <Button onClick={jest.fn()}>Test</Button>
        )
        expect(container.firstChild).toMatchSnapshot()
        expect(getByText('Test')).toBeInTheDocument()
    })
    it('should render filled', () => {
        const { getByText } = render(
            <Button onClick={jest.fn()} filled>
                Test
            </Button>
        )
        expect(getByText('Test')).toHaveClass('filled')
    })
    it('should render inactive', () => {
        const { getByText } = render(
            <Button onClick={jest.fn()} inactive>
                Test
            </Button>
        )
        expect(getByText('Test')).toHaveClass('inactive')
        expect(getByText('Test')).toBeDisabled()
    })
    it('should render with style', () => {
        const { getByText } = render(
            <Button onClick={jest.fn()} style={{ color: 'red' }}>
                Test
            </Button>
        )
        expect(getByText('Test')).toHaveStyle('color: red')
    })
    it('should render with submit type', () => {
        const { getByText } = render(
            <Button onClick={jest.fn()} submit>
                Test
            </Button>
        )
        expect(getByText('Test')).toHaveAttribute('type', 'submit')
    })
    it('should call onClick', () => {
        const onClick = jest.fn()
        const { getByText } = render(<Button onClick={onClick}>Test</Button>)
        fireEvent.click(getByText('Test'))
        expect(onClick).toHaveBeenCalled()
    })
    it('should not call onClick if inactive', () => {
        const onClick = jest.fn()
        const { getByText } = render(
            <Button onClick={onClick} inactive>
                Test
            </Button>
        )
        fireEvent.click(getByText('Test'))
        expect(onClick).not.toHaveBeenCalled()
    })
    it('should not call onClick on enter if inactive', () => {
        const onClick = jest.fn()
        const { getByText } = render(
            <Button onClick={onClick} inactive>
                Test
            </Button>
        )
        fireEvent.keyDown(getByText('Test'), { key: 'Enter', code: 13 })
        expect(onClick).not.toHaveBeenCalled()
    })
})
