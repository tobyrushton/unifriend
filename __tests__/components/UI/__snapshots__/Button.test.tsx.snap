// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  class="fontText button  "
  tabindex="0"
  type="button"
>
  Test
</button>
`;
